lib/src/screens/connect_hotspot_prefix.dart/view_model/connect_hotspot_prefix_view_model.dart

```diff
@@ -147,68 +125,41 @@ class ConnectHotspotPrefixViewModel extends ChangeNotifier {
   }
 
   void connectToHotspot() async {
-    try {
-      debugPrint('connectToHotspot called and retryCount is $_retryCount');
-      _setWrongPassword(false);
-      _setIsAttemptingConnectionToHotspot(true);
-      _setIsRetryingHotspot(false);
-
-      final connectedSSID = await PluginWifiConnect.ssid;
-      debugPrint('Current SSID: $connectedSSID');
-      // In case we are already connected to the hotspot, we can just go to the next step, finding the provisioned machine.
-      if (connectedSSID != null && connectedSSID.startsWith(_hotspotPrefix) && _connectedToHotspot) {
-        debugPrint('Already connected to $_hotspotPrefix hotspot');
+    debugPrint('connectToHotspot called');
+    _setIsAttemptingConnectionToHotspot(true);
+
+    final connectedSSID = await hotspotProvisioningRepository.getCurrentSSID();
+    debugPrint('Current SSID: $connectedSSID');
+    // In case we are already connected to the hotspot, we can just go to the next step, finding the provisioned machine.
+    if (connectedSSID != null && connectedSSID.startsWith(hotspotPrefix) && _connectedToHotspot) {
+      debugPrint('Already connected to $hotspotPrefix hotspot');
+      _findProvisionedMachine();
+      return;
+    }
+    // If we are not connected to the hotspot, we need to connect to it.
+    debugPrint('Connecting to $hotspotPrefix-#### hotspot');
+    final connected = await hotspotProvisioningRepository.connectToSecureNetworkByPrefix(
+      prefix: hotspotPrefix,
+      password: hotspotPassword,
+      isWep: false,
+      isWpa3: false,
+      saveNetwork: true, // flips joinOnce on iOS to false
+    );
+    if (connected == true) {
```

Comment:
```suggestion
    if (connected) 
```
now that this can't be null you can simplify^